src/Microsoft.Azure.WebJobs.Host/Queues/StorageQueueMessageExtensions.cs

```diff
@@ -27,6 +27,10 @@ public static string TryGetAsString(this IStorageQueueMessage message)
                 // when the message type is binary, AsString will throw if the bytes aren't valid UTF-8.
                 value = null;
             }
+            catch (FormatException)
```

Comment:
Exception filters clean the proposed code, but we'd need to be able to use C# 6 in the project :)